### PR TITLE
Fix Aha waiters for replay and check mirror status (SYN-3923)

### PR DESCRIPTION
**Pull request by mikemoritz**

### CODE REVIEW

File: synapse/lib/nexus.py [hunk 1]
```diff
@@ -465,6 +467,7 @@ async def runMirrorLoop(self, proxy):
                     # with tellready we move to ready=true when we get a None
                     if item is None:
                         await self.setNexsReady(True)
+                        self.mirready.set()
```
Comment (vEpiphyte) [inline synapse/lib/nexus.py:470]: Should this get cleared in the event of a follower -> leader promotion?

Comment (mikemoritz) [reply synapse/lib/nexus.py:472]: I was thinking of this as very targeted for the test case, but easy enough to clear it on startup() which also gets called from promote()

Comment (vEpiphyte) [reply synapse/lib/nexus.py:472]: If we could mark it as a `_` prefixed event so others don't possibly rely on it?

Comment (vEpiphyte) [reply synapse/lib/nexus.py:472]: Also would breaking out of the loop (for example, in a reconnect scenario) also trigger clearing the event?

Comment (mikemoritz) [reply synapse/lib/nexus.py]: updated

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**